src/natcap/invest/urban_nature_access.py

```diff
@@ -459,6 +517,147 @@ def execute(args):
     LOGGER.info('Finished Urban Nature Access Model')
 
 
+def _sum_rasters(greenspace_supply_list, target_sum_path):
+    """Sum a stack of rasters.
+
+    This assumes that all rasters have the same nodata, equal to
+    ``FLOAT32_NODATA``.
+
+    Args:
+        greenspace_supply_list (list): A list of float32 raster paths, where
+            pixel values in these rasters will be summed along the 3rd
+            dimension.  Rasters in this stack must all perfectly align.
+        target_sum_path (string): The target path.
+
+    Returns:
+        ``None``
+    """
+    def _sum_op(*greenspace_supply_arrays):
+        """Sum a stack of arrays that all use FLOAT32_NODATA as their nodata.
+
+        Returns a float32 numpy array.
+        """
+        valid_mask = numpy.ones(greenspace_supply_arrays[0].shape, dtype=bool)
+        for array in greenspace_supply_arrays:
+            valid_mask &= ~utils.array_equals_nodata(array, FLOAT32_NODATA)
+
+        sum_array = numpy.zeros(valid_mask.shape, dtype=numpy.float32)
+        for array in greenspace_supply_arrays:
+            sum_array[valid_mask] += array[valid_mask]
+        result = numpy.full(valid_mask.shape, FLOAT32_NODATA,
+                            dtype=numpy.float32)
+        result[valid_mask] = sum_array[valid_mask]
+        return result
+
+    pygeoprocessing.raster_calculator(
+        [(path, 1) for path in greenspace_supply_list], _sum_op,
+        target_sum_path, gdal.GDT_Float32, FLOAT32_NODATA)
+
+
+def _reclassify_greenspace_area(
+        lulc_raster_path, lulc_attribute_table, target_raster_path,
+        only_these_greenspace_codes=None):
+    """Reclassify LULC pixels into the greenspace area they represent.
```

Comment:
This is a little unclear to me. Maybe another line saying something like "Greenspace pixels' value will be the pixel area, and non-greenspace pixels' value will be 0." (If I'm understanding that correctly)

Reply:
Added a description along these lines in 0dacced0e